pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
```

Comment:
I think the check should be near it's access. Not further.
If we do the check in the caller, the function needs to assume the check was made elsewhere, and everywhere we call the function we have to remember to check the pointer. End result? More panics in the future when we forget this implicit precondition to call `updateImageTriggers`.

Look how we check other pointers in the spot before access: https://github.com/openshift/origin/pull/4505/files#diff-82f54c5f4bf72a64f869a69e557df8cbL215, https://github.com/openshift/origin/pull/4505/files#diff-82f54c5f4bf72a64f869a69e557df8cbL218